bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -323,9 +364,8 @@ fn main() -> Result<()> {
         }
 
         if shutdown {
-            logs_agent.shutdown();
-            dogstats_client.shutdown();
-            telemetry_listener.shutdown();
+            dogstats_cancel_token.cancel();
+            telemetry_listener_cancel_token.cancel();
```

Comment:
should this be a join/wait for the flush to finish rather than a canceling the flush?

Reply:
canceling stops the main loop tasks, not any of the running tasks to aggregate/flush data. So cancelling first then flushing should be okay. We will test and confirm.

Reply:
Yeah main _desperately_ needs automated tests, you're completely right.

I'm HOPING that we can carve out time to emulate the telemetry API (and maybe share it with the rest of the community?), which would enable a really good e2e test for main.